src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -1597,6 +1597,31 @@ public void TestEvent()
             Assert.Equal(0, activity.Events.ElementAt(1).Tags.Count());
         }
 
+        [Fact]
+        public void AddLinkTest()
+        {
+            ActivityContext c1 = new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), ActivityTraceFlags.None);
+            ActivityContext c2 = new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), ActivityTraceFlags.None);
+
+            ActivityLink l1 = new ActivityLink(c1);
+            ActivityLink l2 = new ActivityLink(c2, new ActivityTagsCollection()
+            {
+                new KeyValuePair<string, object?>("foo", 99)
+            });
+
+            Activity activity = new Activity("LinkTest");
+            Assert.True(ReferenceEquals(activity, activity.AddLink(l1)));
+            Assert.True(ReferenceEquals(activity, activity.AddLink(l2)));
```

Comment:
Do we need a test that validates what happens (which presumably means "doesn't blow up") if the same ActivityLink is added twice?

Reply:
@antonfirsov Also, could you please add some more tests inside the test method [EnumerateLinksTest()](https://github.com/dotnet/runtime/blob/40bc2d88b6606324f5774cc972d480a1d26084f8/src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs#L2143)?

Reply:
Fixed in [b367037](https://github.com/dotnet/runtime/pull/101381/commits/b367037c92a916bab076127e8fad3c1df78a3d0e).